src/core/display/Container.js

```diff
@@ -628,11 +628,20 @@ Container.prototype.destroy = function (destroyChildren)
 {
     DisplayObject.prototype.destroy.call(this);
 
-    if (destroyChildren)
-    {
-        for (var i = 0, j = this.children.length; i < j; ++i)
-        {
-            this.children[i].destroy(destroyChildren);
+    if (!destroyChildren) {
+    } else {
+        for (var i = 0, j = this.children.length; i < j; ++i) {
+            var child = this.children[i];
+            var isSprite = !!child._texture; //TODO: Need a better way to detect if it's a sprite
```

Comment:
Yeah I don't think we can, which is why this idea of checking here isn't great. A parent knowing about how its children override a function is a crazy concept in OOP.

Ideally we would fix the discrepency, but changing this behavior (or the function signature) is a breaking change which is why I haven't done it yet :(